### ISSUE TITLE
undefined method `children' for :private_constant:Symbol (NoMethodError)

### ISSUE THREAD

**Issue by gareth** [castwide/solargraph#408]
I noticed this via the VSCode extension (v0.21.1) but was able to reproduce it directly in the gem.

**Versions:**
Gem: 0.40.2, 0.39.17
Ruby: 2.6.5
OS: Mac OS X 10.15.7 (19H114)

When trying to use Solargraph on the ruby (2.6.5) core lib folder, this exception is raised:
> undefined method `children' for :private_constant:Symbol (NoMethodError)

This also means the folder is unexplorable in e.g. VSCode with the Solargraph extension.

This appears related to #312 which referenced a similar problem with use of `private_class_method`.

```
ruby/2.6.0$ pwd
/usr/local/var/rbenv/versions/2.6.5/lib/ruby/2.6.0
ruby/2.6.0$ solargraph scan
[WARN] Error loading core pin cache: [ArgumentError] undefined class/module Solargraph::Pin::YardPin::
Traceback (most recent call last):
	50: from /usr/local/var/rbenv/versions/2.6.5/bin/solargraph:23:in `<main>'
	49: from /usr/local/var/rbenv/versions/2.6.5/bin/solargraph:23:in `load'
	48: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/bin/solargraph:5:in `<top (required)>'
	47: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor/base.rb:485:in `start'
	46: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
	45: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command'
	44: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
	43: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/shell.rb:172:in `scan'
	42: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/benchmark-0.1.0/lib/benchmark.rb:293:in `measure'
	41: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/shell.rb:173:in `block in scan'
	40: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/api_map.rb:176:in `load'
	39: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/api_map.rb:63:in `catalog'
	38: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/api_map.rb:63:in `each'
	37: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/api_map.rb:83:in `block in catalog'
	36: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/source_map.rb:156:in `map'
	35: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/source_map/mapper.rb:50:in `map'
	34: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/source_map/mapper.rb:26:in `map'
	33: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/rubyvm/class_methods.rb:29:in `map'
	32: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor.rb:38:in `process'
	31: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:37:in `process'
	30: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:47:in `process_children'
	29: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:47:in `each'
	28: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:49:in `block in process_children'
	27: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor.rb:38:in `process'
	26: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/rubyvm/node_processors/namespace_node.rb:33:in `process'
	25: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:47:in `process_children'
	24: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:47:in `each'
	23: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:49:in `block in process_children'
	22: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor.rb:38:in `process'
	21: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/rubyvm/node_processors/scope_node.rb:9:in `process'
	20: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:47:in `process_children'
	19: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:47:in `each'
	18: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:49:in `block in process_children'
	17: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor.rb:38:in `process'
	16: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:37:in `process'
	15: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:47:in `process_children'
	14: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:47:in `each'
	13: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:49:in `block in process_children'
	12: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor.rb:38:in `process'
	11: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:37:in `process'
	10: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:47:in `process_children'
	 9: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:47:in `each'
	 8: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:49:in `block in process_children'
	 7: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor.rb:38:in `process'
	 6: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:37:in `process'
	 5: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:47:in `process_children'
	 4: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:47:in `each'
	 3: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor/base.rb:49:in `block in process_children'
	 2: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/node_processor.rb:38:in `process'
	 1: from /usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/rubyvm/node_processors/send_node.rb:32:in `process'
/usr/local/var/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/solargraph-0.40.2/lib/solargraph/parser/rubyvm/node_processors/send_node.rb:230:in `process_private_constant': undefined method `children' for :private_constant:Symbol (NoMethodError)
```

**Comment by castwide** [comment]
Fixed in v0.40.3.

**Comment by gareth** [comment]
Thanks for the fix! I didn't notice that there was already a to-do comment so I'm glad it wasn't a big unknown
